server/service/src/report/report_service.rs

```diff
@@ -162,17 +162,81 @@ pub trait ReportServiceTrait: Sync + Send {
         let file = file_service
             .store_file(
                 &format!("{}_{}.html", now.format("%Y%m%d_%H%M%S"), report_name),
-                format!(
-                    "{}{}{}",
-                    document.header.unwrap_or("".to_string()),
-                    document.document,
-                    document.footer.unwrap_or("".to_string())
-                )
-                .as_bytes(),
+                self.format_html_document(document).as_bytes(),
             )
             .map_err(|err| ReportError::DocGenerationError(format!("{}", err)))?;
         Ok(file.id)
     }
+
+    fn format_html_document(&self, document: GeneratedReport) -> String {
+        format!(
+            "<html>
+  <head>
+    <style>
+      @page {{
+        size: A4;
+        margin: 1.5cm;
```

Comment:
it's the fixed margins that I'm not happy about. apart from that, the approach is pretty neat.

Reply:
Could it be configurable in the future? or is there a reason for 1.5cm

btw @ujwalSussol discovered that the header is limited to 50px on headless chrome...

Reply:
Is the page numbering done automatically?